contrib/systemd/mako.service.in

```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=A lightweight Wayland notification daemon
+Documentation=man:mako(1)
+PartOf=wayland-session.target
```

Comment:
There's no such target upstream, so this requires user to have a lot of local glue for this to work. [There's already an existing `graphical-session.target` for wayland sessions which should be used](https://www.freedesktop.org/software/systemd/man/systemd.special.html).

Reply:
this has been also discussed above
https://github.com/emersion/mako/pull/188#discussion_r322722986

the problem with graphical-session.target is of-course when you run in a Xorg environment and then these wayland services fail.

Reply:
See https://github.com/emersion/mako/pull/240#issuecomment-597122444

Reply:
@gdamjan I don't think that's a scenario we realistically need to address.

How many users do we expect to install mako, and then run Xorg?
Even for this already unusual scenario, mako will simply fail to start up and the service stop.

Reply:
Count me was one. I was evaluating `sway` with `systemd` but ran into problems, so I reverted to `i3` and XOrg, so yes, I potentially have all those packages installed on my system. I may try `sway` with `systemd` again in the future, so I don't necessarily want to uninstall them. So the `wayland-session.target` makes sense to me.

Reply:
If you're running i3 and have mako installed now, `mako` should execute when a notification is sent (via dbus activation).

Whatever steps you've taken to prevent that happening on your current setup, will also continue working if this unit says `graphical-session.target`.

---

I *really* don't want to depend on a non-standard target that one has to keep locally. The whole point of upstreaming this service file, is to standardise it and not have to track it in my dotfiles (and anyone can rely on it too).

By tying it to a non-stardard target, I still have to track a custom file locally just for this, and any other user has to replicate that, making it all pointless. `systemd` provides a target specifically designed for this use case, I'm really against using a separate, custom, one.

Reply:
I was using the `sway-gnome` project. I checked what they are doing for the `mako` systemd file, which you can see here:

https://github.com/Drakulix/sway-gnome/blob/master/systemd/user/mako.service

They depend on `sway-session.target` which is provided by the same package, so it's self-contained. That also explains why I didn't end up running `mako` with XOrg.

Reply:
So, let me explain this again, because I'm trying to explain that using `graphical-session.target` should not be a problem for your setup anyway.

Mako uses dbus-activation, which means that it'll get triggered by any attempt at sending a notification. Currently there is no systemd service file. As such, it just runs.

You probably have _something_, whatever it is, that avoids mako starting.

On the other side, we want to introduce `PartOf=graphical-session.target`. What this means is that when `graphical-session.target`, mako gets killed. It does not affect its starting condition. It'll only change from running unmanaged, to running managed by systemd.

The _something_ I mentioned above that keeps mako from running on your system will still prevent it from running after this change.

Relevant (from systemd.unit(5)):
```
       PartOf=
           Configures dependencies similar to Requires=, but limited to stopping and restarting of units. When systemd
           stops or restarts the units listed here, the action is propagated to this unit. Note that this is a one-way
           dependency — changes to this unit do not affect the listed units.

           When PartOf=b.service is used on a.service, this dependency will show as ConsistsOf=a.service in property
           listing of b.service.  ConsistsOf= dependency cannot be specified directly.
```

My suggestion is that you install this systemd unit file, restart, and check that nothing breaks for you. I'm fairly confident nothing will change (assuming you don't explicitly enable the service file, of course).